File: 프로그래머스/0/181912. 배열 만들기 5/배열 만들기 5.py
def solution(intStrs, k, s, l):
    answer = []
    
    for st in intStrs:
        # 1. s번 인덱스부터 길이 l만큼 잘라내서 정수로 변환
        num = int(st[s : s + l])
        
        # 2. 변환한 값이 k보다 큰 경우에만 결과 배열에 추가
        if num > k:
            answer.append(num)
            
    return answer

File: 프로그래머스/0/181912. 배열 만들기 5/배열 만들기 5.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<string> intStrs, int k, int s, int l) {
    vector<int> answer;
    
    for (int idx = 0; idx < intStrs.size(); idx++) {
        int parsed_num = 0;
        
        // s부터 s + l - 1 까지 (길이 l 만큼) 앞에서부터 읽기
        for (int j = s; j < s + l; j++) {
            // 기존 값에 10을 곱하고 새 자릿수를 더함 (아스키 '0' 차감)
            parsed_num = parsed_num * 10 + (intStrs[idx][j] - '0');
        }
        
        // 완결된 정수 하나를 k와 비교하여 push
        if (parsed_num > k) {
            answer.push_back(parsed_num);
        }
    }
    
    return answer;
}